src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,150 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int defaultCapacity = 16;
+    private int size;
+    private int capacity = defaultCapacity;
+    private final double loadFactor = 0.75;
+    private int treshold = (int)(capacity * loadFactor);
+    private Node[] table = new Node[capacity];
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > treshold) {
+            resize();
+        }
+        for (Node element : table) {
+            if (element != null && (element.getValue().equals(value)
+                    || element.getValue() == value)) {
+                return;
+            }
+            if (element != null && element.getNext() != null) {
+                while (element != null) {
+                    if (element.getValue() == value || element.getValue().equals(value)) {
+                        return;
+                    }
+                    element = element.getNext();
+                }
+            }
+        }
+        if (key == null || indexForElement(hash(key),capacity) == 0) {
+            addToFirstPosition(key,value);
```

Comment:
why do we need a separate method to add at first position? Lets reformat method indexForElement instead